cmd/minishift/cmd/openshift/registry.go

```diff
@@ -46,7 +47,7 @@ var registryCmd = &cobra.Command{
 
 		util.ExitIfNotRunning(host.Driver, constants.MachineName)
 
-		registryInfo, err := openshift.GetDockerRegistryInfo()
+		registryInfo, err := openshift.GetDockerRegistryInfo(addon.GetAddOnManager())
```

Comment:
Just pass 'addOnManager.Get("registry-route")' to openshift.GetDockerRegistryInfo. This is better encapsulation and it most likely will fix your dependency issues.